### DISCUSSION TITLE
How to make a popup window in my own code?

### DISCUSSION THREAD

**Discussion by schulle4u** [datenstrom/community#301] [Ask a question] [answered]
Happy new year! 

Currently I'm trying to extend my audio plugin and would like to have an optional popup player for audio files, useful to play back larger files or streams without interruption while browsing the rest of the page. So far I've discovered two methods which in theory can achieve this. 

First one: Create a standalone HTML page with the audio player in it, and call this page in a `window.open()` JS function. This however seems not easily possible as there is no way to hand over the needed parameters from the Yellow plugin. At least `$audiourl` is needed. 

Second method is to include the needed HTML into a JS function. I found [this very basic example code](https://www.w3schools.com/jsref/tryit.asp?filename=tryjsref_win_open3) from W3schools and successfully was able to print their example text in a popup window. However, as soon as I try to extend it with the needed player code, the popup window doesn't work anymore. 

```
				$output .="<script>
function playerPopup() {
	var playerWindow = window.open(\"\", \"Audio\", \"width=200,height=100\");
	
	// This doesn't work
	playerWindow.document.write(\"<p><audio src=\"".htmlspecialchars($audiourl)."\" controls autoplay>HTML5 audio not supported.</audio></p>\";

	// The example output works
	// playerWindow.document.write(\"<p>This is 'MsgWindow'. I am 200px wide and 100px tall!</p>\");
}
</script>\n";
```

The problem seems not to be on yellow's side as the correct content of `$audiourl` is in the HTML/JS output. Is there another solution I'm currently missing? Not just on the javascript part, perhaps good old yellow offers something I wasn't thinking of yet. 

Steffen

**Comment by markseuffert** [comment, marked as answer]
Happy new year, Steffen.

Can you upload your source code? You might get more answers with a minimal, easily runnable, and well-formatted code that illustrates your problem. If you want feedback for general programming questions, also have a look at [Stackoverflow](http://stackoverflow.com) and [Codekicker](http://codekicker.de).

**Comment by schulle4u** [comment]
Thanks for the feedback. OK since this seems not to be directly yellow-related, I'll try to find a general solution and adapt it to my needs. In the meanwhile I've found various solutions on stackoverflow, but nothing that fully satisfies my imaginations.
